azurerm/internal/services/subscription/subscription_resource.go

```diff
@@ -0,0 +1,472 @@
+			return fmt.Errorf("could not update Display Name of Subscription %q: %+v", *subscriptionId, err)
+		}
+	}
+
+	return nil
+}
+
+func resourceSubscriptionRead(d *schema.ResourceData, meta interface{}) error {
+	aliasClient := meta.(*clients.Client).Subscription.AliasClient
+	client := meta.(*clients.Client).Subscription.Client
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.SubscriptionAliasID(d.Id())
+	if err != nil {
+		return err
+	}
+	d.Set("alias", id.Name)
+
+	alias, err := aliasClient.Get(ctx, id.Name)
+	if err != nil {
+		if utils.ResponseWasNotFound(alias.Response) {
+			log.Printf("[INFO] Error reading Subscription %q - removing from state", d.Id())
+			d.SetId("")
+			return nil
+		}
+
+		return fmt.Errorf("reading Subscription Alias %q: %+v", id.Name, err)
+	}
+
+	subscriptionId := ""
+	subscriptionName := ""
+	tenantId := ""
+	t := make(map[string]*string, 0)
+	if props := alias.Properties; props != nil && props.SubscriptionID != nil {
+		subscriptionId = *props.SubscriptionID
+		resp, err := client.Get(ctx, subscriptionId)
+
+		if err != nil {
+			return fmt.Errorf("failed to read Subscription %q (Alias %q) for Tenant Information: %+v", subscriptionId, id.Name, err)
+		}
+		if resp.TenantID != nil {
+			tenantId = *resp.TenantID
+		}
+
+		if resp.DisplayName != nil {
+			subscriptionName = *resp.DisplayName
+		}
+
+		t = resp.Tags
+	}
+
+	// (@jackofallops) A subscription's billing scope is not exposed in any way in the API/SDK so we cannot read it back here
+
+	d.Set("subscription_id", subscriptionId)
+	d.Set("subscription_name", subscriptionName)
+	d.Set("tenant_id", tenantId)
+	tags.FlattenAndSet(d, t)
+
+	return nil
+}
+
+// (@jackofallops) - Delete here is a misnomer.  The nature of subscriptions is such that they are never truly deleted
+// Deleted here means Cancelled. Cancelled subscriptions are held in this state for  90 days before being purged from
+// active systems.  However, the backend billing systems _never_ remove this data, so once a Subscription ID has been
+// used and purged from active use it can never be recovered nor the UUID reused.
+// Note Cancelling a Subscription leaves it in one of several states, `Disabled` for a Subscription with no Resources or
+// Alias assignments, `Warned` for Cancelled with "something" associated with it.
+func resourceSubscriptionDelete(d *schema.ResourceData, meta interface{}) error {
+	aliasClient := meta.(*clients.Client).Subscription.AliasClient
+	subscriptionClient := meta.(*clients.Client).Subscription.SubscriptionClient
+	client := meta.(*clients.Client).Subscription.Client
+	ctx, cancel := timeouts.ForDelete(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.SubscriptionAliasID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	locks.ByName(id.Name, SubscriptionResourceName)
+	defer locks.ByName(id.Name, SubscriptionResourceName)
```

Comment:
(same here) do we want to be locking on the Subscription ID?

Reply:
As above.